src/python/bot/fuzzers/mutator_plugin.py

```diff
@@ -0,0 +1,155 @@
+  assert storage.copy_file_from(url, file_path)
+  return file_path
+
+
+def _unpack_mutator_plugin(mutator_plugin_archive_path):
+  """Unpacks |mutator_plugin_archive_path| in the unpacked plugins directory and
+  returns the path it was unpacked into."""
+  mutator_plugin_name = os.path.basename(
+      os.path.splitext(mutator_plugin_archive_path)[0])
+  unpacked_plugin_dir = os.path.join(_get_mutator_plugins_unpacked_dir(),
+                                     mutator_plugin_name)
+  archive.unpack(mutator_plugin_archive_path, unpacked_plugin_dir)
+  return unpacked_plugin_dir
+
+
+def _extract_name_from_archive(plugin_archive_filename):
+  """Parses |plugin_archive_filename| which should be named using the schema:
+  '$NAME-$JOB-$FUZZ_TARGET.zip'. Returns tuple containing
+  ($NAME, $JOB-$FUZZ_TARGET)."""
+  # TODO(metzman): Get rid of this when we create an upload page for custom
+  # mutator plugins.
+  plugin_archive_name = os.path.splitext(plugin_archive_filename)[0]
+  idx = plugin_archive_name.index('-')
+  plugin_name = plugin_archive_name[:idx]
+  return plugin_name, plugin_archive_name[idx + 1:]
+
+
+class PluginGetter(object):
+  """Class that gets a usable mutator plugin for |fuzzer_binary_name| from
+  GCS."""
+
+  def __init__(self, fuzzer_binary_name):
+    self.fuzzer_binary_name = fuzzer_binary_name
+    self.job_name = environment.get_value('JOB_NAME')
+    self.create_directories()
+
+  def create_directories(self):
+    """Creates directories needed to use mutator plugins."""
+    plugins_root_dir = environment.get_value('MUTATOR_PLUGINS_DIR')
+    assert plugins_root_dir
+    shell.remove_directory(plugins_root_dir, recreate=True)
+    os.mkdir(_get_mutator_plugins_archives_dir())
+    os.mkdir(_get_mutator_plugins_unpacked_dir())
+
+  def _is_plugin_usable(self, plugin_archive_filename):
+    """Returns True if |plugin_archive_filename| is a usable plugin for this
+    job, fuzz target combination."""
+    _, plugin_job_and_fuzzer = _extract_name_from_archive(
+        plugin_archive_filename)
+    expected_name = '%s-%s' % (self.job_name, self.fuzzer_binary_name)
+    return expected_name == plugin_job_and_fuzzer
+
+  def get_mutator_plugin(self):
+    """Downloads, and unpacks a usable mutator plugin for this job and fuzz
+    target if one is available in GCS"""
+    mutator_plugins = _get_mutator_plugins_from_bucket()
+    usable_mutator_plugins = [
+        plugin_archive for plugin_archive in mutator_plugins
+        if self._is_plugin_usable(plugin_archive)
+    ]
+
+    # Quit if no usable plugins are available.
+    if not usable_mutator_plugins:
+      return None
+
+    plugin_archive_name = utils.random_element_from_list(usable_mutator_plugins)
+    plugin_archive_path = _download_mutator_plugin_archive(plugin_archive_name)
+    _unpack_mutator_plugin(plugin_archive_path)
+    mutator_plugin_path = find_mutator_plugin()
+    assert mutator_plugin_path is not None
```

Comment:
We should gracefully exit with log_error, and return bool result rather than assert.